src/ngraph/runtime/reference/clamp.hpp

```diff
@@ -0,0 +1,66 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <cmath>
+#include <cstddef>
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            template <typename T>
+            T double_to_T(double x, double float_to_int_converter(double))
+            {
+                if (std::is_integral<T>())
+                {
+                    double min_t = static_cast<double>(std::numeric_limits<T>::min());
+                    double max_t = static_cast<double>(std::numeric_limits<T>::max());
+                    x = std::max(x, min_t);
+                    x = std::min(x, max_t);
+                    x = float_to_int_converter(x);
+                }
+                return static_cast<T>(x);
+            }
+
+            template <typename T>
+            void clamp(const T* arg, T* out, double min, double max, size_t count)
+            {
+                for (size_t i = 0; i < count; i++)
+                {
+                    T min_t = double_to_T<T>(min, [](double x) { return std::ceil(x); });
+                    T max_t = double_to_T<T>(max, [](double x) { return std::floor(x); });
```

Comment:
It should be an error if the value of min/max can't exactly convert to integers.

Reply:
There are two potential issues when converting from min/max attributes (type double) to type T (matching the type of the argument) when T is an integer.
1. The attribute my not have an integral value
2. The attribute value may be greater than the maximum or less than the minimum value that can be represented by the type T

My solutions:
1. Provide a lambda to convert the attribute to an integral value
2. Return numeric_limits<T>::max() or min() respectively

No errors.  Note that this conversion (call to double_to_int) is now made from src/ngraph/op/fused/clamp.cpp instead of the reference kernel.